### ISSUE TITLE
Fatal Error when editing event participant w/Multi-domain

### ISSUE THREAD

**Issue by darrick** [iATSPayments/com.iatspayments.civicrm#151]
I have CiviCRM under three domains.  Under domain 1 I have no Payment Processor for ACH/EFT payments.  Under domain 3 I do have a ACH/EFT processor. 

When I go to view or edit a event participant under domain 1 I get a Fatal Error:

"'11' is not a valid option for field payment_processor"

ID: 11 is the ACH/EFT payment processor for domain 3.  

The error is at line 946 iats.php.

I chatted with Eileen and she said probably best to wrap it in a try ... catch.  Other option is to send domain_id to the iats_civicrm_processors function.

**Comment by adixon** [comment]
Yeah, there are still a bunch of api calls not wrapped in try/catches.

I haven't been paying attention to domain_id, but I think the domain_id should probably be implicit in the iats_civicrm_processors function, unless there are times where a domain_id agnostic call should be made ...
